src/main/java/io/jenkins/docker/connector/DockerComputerJNLPConnector.java

```diff
@@ -76,46 +101,80 @@ public boolean isLaunchSupported() {
 
             @Override
             public void launch(SlaveComputer computer, TaskListener listener) throws IOException, InterruptedException {
-                final DockerClient client = api.getClient();
-
-                List<String> args = buildCommand(workdir, computer);
+                final DockerContainerLifecycleHandler handler = new DockerContainerLifecycleHandler(){
+                    @Override
+                    public void beforeContainerCreated(DockerAPI api, String workdir, CreateContainerCmd cmd) throws IOException, InterruptedException {
+                        if(!passSlaveConnectionArgs) {
+                            ensureWaiting(cmd);
+                        }
+                    }
+
+                    @Override
+                    public void afterContainerStarted(DockerAPI api, String workdir, String containerId) throws IOException, InterruptedException {
+                        if(!passSlaveConnectionArgs) {
+                            final DockerClient client = api.getClient();
+                            injectRemotingJar(containerId, workdir, client);
+                        }
+                    }
+                };
+                CreateContainerCmd cmd = template.createContainerCmd(api);
+                if(!passSlaveConnectionArgs){
+                    launchAndInjectJar(computer, listener, handler, api, containerExecuter, workdir, cmd);
+                }else{
+                    launchStandardJNLPImage(computer, listener, handler, api, containerExecuter, workdir, cmd);
```

Comment:
just replace the exec-based implementation with `java remoting.jar` set as main process in container